graph/db/sql_store.go

```diff
@@ -80,3 +131,749 @@ func NewSQLStore(db BatchedSQLQueries, kvStore *KVStore,
 
 	return s, nil
 }
+
+// TxOptions defines the set of db txn options the SQLQueries
+// understands.
+type TxOptions struct {
+	// readOnly governs if a read only transaction is needed or not.
+	readOnly bool
+}
+
+// ReadOnly returns true if the transaction should be read only.
+//
+// NOTE: This implements the TxOptions.
+func (a *TxOptions) ReadOnly() bool {
+	return a.readOnly
+}
+
+// NewReadTx creates a new read transaction option set.
+func NewReadTx() TxOptions {
+	return TxOptions{
+		readOnly: true,
+	}
+}
+
+// AddLightningNode adds a vertex/node to the graph database. If the node is not
+// in the database from before, this will add a new, unconnected one to the
+// graph. If it is present from before, this will update that node's
+// information.
+//
+// NOTE: part of the V1Store interface.
+func (s *SQLStore) AddLightningNode(node *models.LightningNode,
+	opts ...batch.SchedulerOption) error {
+
+	ctx := context.TODO()
+
+	r := &batch.Request[SQLQueries]{
+		Opts: batch.NewSchedulerOptions(opts...),
+		Do: func(queries SQLQueries) error {
+			_, err := upsertNode(ctx, queries, node)
+			return err
+		},
+	}
+
+	return s.nodeScheduler.Execute(ctx, r)
+}
+
+// FetchLightningNode attempts to look up a target node by its identity public
+// key. If the node isn't found in the database, then ErrGraphNodeNotFound is
+// returned.
+//
+// NOTE: part of the V1Store interface.
+func (s *SQLStore) FetchLightningNode(pubKey route.Vertex) (
+	*models.LightningNode, error) {
+
+	ctx := context.TODO()
+
+	var (
+		readTx = NewReadTx()
+		node   *models.LightningNode
+	)
+	err := s.db.ExecTx(ctx, &readTx, func(db SQLQueries) error {
+		var err error
+		_, node, err = getNodeByPubKey(ctx, db, pubKey)
+
+		return err
+	}, func() {})
```

Comment:
Can we add a `sqldb.NoOpReset` and use that here? Then we can more easily find and replace them once we turn this into a functional option in sqldbv2.